File: 0001 - Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> our_map;
        vector<int> result;
        for(int i = 0; i < nums.size(); i++){
            int difference = target - nums[i];
            if(our_map.find(difference) != our_map.end()){
                result.push_back(i);
                result.push_back(our_map[difference]);
                return result;
            }else our_map[nums[i]] = i;
        }
        return result;
    }
};

File: 0001 - Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        our_map = {}
        for index, num in enumerate(nums):
            difference = target - num
            if difference in our_map:
                return [our_map[difference], index]
            our_map[num] = index